ddsink/handler.py

```diff
@@ -0,0 +1,60 @@
+import datetime
+import gzip
+import json
+import logging
+import os
+import uuid
+from typing import Iterator
+
+import boto3
+
+from ddsink.metrics import (
+    initialize_datadog,
+    load_config,
+    query_datadog,
+    format_metric_results,
+    MetricConfig,
+)
+from ddsink.types import JSONDict
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def save_output_to_s3(config: MetricConfig, output: Iterator[JSONDict])-> None:
+    dt = datetime.datetime.fromtimestamp(config.end_time)
+    file_name = config.s3_path.format(
```

Comment:
Totally unnecessary here but FYI: format strings can drive all the sub formatting from themselves now, e.g.:

'date={dt:%Y-%m-%d}/hour={dt:%H}'.format(dt=dt)